### PR TITLE
Split deployment.yaml config sample RBAC roles

### PR DESCRIPTION
**Pull request by manfontan**
# Description

Because https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1727 the operator does not work if the deployment doesn't have RBAC access to listing nodes. This PR mitigates this issue for users relying on deployment config sample to build their cluster manifests.

The deployment config sample uses a RoleBinding pointing to a ClusterRole. This configuration will not allow to list nodes.

In this PR we split the ClusterRole in two ClusterRoles one to access namespace scoped resources and another to access cluster-wide (nodes) resources. Using a RoleBinding for the first and a ClusterRoleBinding for the former.

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

**Review by johscheuer — CHANGES_REQUESTED** [review]

File: config/samples/deployment.yaml [hunk 1]
```diff
@@ -132,6 +124,35 @@ subjects:
 - kind: ServiceAccount
   name: fdb-kubernetes-operator-controller-manager
 ---
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (manfontan) [reply config/samples/deployment.yaml:142]: 🤔 ok looks like roles.yaml is also generated using https://book.kubebuilder.io/reference/markers/rbac.html
But it does not support adding multiple ClusterRoles.
https://github.com/FoundationDB/fdb-kubernetes-operator/blob/d21d0f37f99c83ee228bd08a7eaaa28becf801a1/controllers/cluster_controller.go#L81
Any suggestions?

[the remark this answers is not in the document]
Comment (manfontan) [reply config/samples/deployment.yaml:142]: I can add a cluster_role.yaml and remove the `kubebuilder:rbac` entry for `nodes`. 
Not super elegant but it does the job.

File: config/rbac/cluster_role.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name: manager-clusterrole
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (johscheuer) [inline config/rbac/cluster_role.yaml:15]: Can we add a newline here?

File: config/rbac/kustomization.yaml [hunk 3]
```diff
@@ -2,3 +2,4 @@ kind: Kustomization
 apiVersion: kustomize.config.k8s.io/v1beta1
 resources:
 - role.yaml
+- cluster_role.yaml
```
Comment (johscheuer) [inline config/rbac/kustomization.yaml:5]: Can we add a newline here?

File: config/samples/deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (johscheuer) [reply config/samples/deployment.yaml:142]: This seems like a reasonable approach, given the limitations from Kubebuilder.

**Review by johscheuer — APPROVED** [review]
LGTM 👍

**Review by johscheuer — APPROVED** [review]
Sorry for the delay, changes LGTM 👍

**Pull request merged**